CRM/EventsExtras/Hook/BuildForm/BaseEvent.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+use CRM_EventsExtras_ExtensionUtil as E;
+use CRM_EventsExtras_SettingsManager as SettingsManager;
+
+/**
+ * Abstract class for BuildForm Hook
+ */
+abstract class CRM_EventsExtras_Hook_BuildForm_BaseEvent {
+
+  /**
+   * Event tab to display on the form 
+   */
+  protected $eventTab;
+
+  /**
+   * Constractor for BuildForm class
+   * 
+   * @param string $eventTab
+   * 
+   */
+  protected function __construct($eventTab){
+    $this->eventTab = $eventTab;
+    $this->addEventTabTemplate();
+  }
+
+  /**
+   * Hides options on the localisation page
+   *
+   * @param string $formName
+   * @param CRM_Core_Form $form
+   */
+  abstract function handle($formName, &$form);
+
+  /**
+   * Checks if the hook should be handled.
+   *
+   * @param string $formName
+   * @param class $formClass
+   *
+   * @return bool
+   */
+  protected function shouldHandle($formName, $formClass) {
+    if ($formName === $formClass) {
+      return TRUE;
+    }
+    
+    return FALSE;
+  }
+
+  /**
+   * fuction to hide fields based on settings
+   *
+   * @param array $form
+   *
+   */
+  protected function hideField(CRM_Event_Form_ManageEvent &$form){
+    $configFields = SettingsManager::getConfigFields($this->eventTab);
+    $settingsValue = SettingsManager::getSettingsValue();
+    $hiddenFields = [];
+    foreach ($configFields as $config) {
+     if ($settingsValue[$config['name']] == 0){
+        if  ($config['extra_attributes']['section'] == $this->eventTab){
```

Comment:
We don't need this check again since we are sure that the config fields being fetched already belong to the tab passed in